pkg/cluster/awsiam_test.go

```diff
@@ -0,0 +1,181 @@
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "eksa-unit-test",
+							Namespace: "default",
+						},
+						Spec: anywherev1.AWSIamConfigSpec{
+							AWSRegion:   "test-region",
+							BackendMode: []string{"mode1", "mode2"},
+							MapRoles: []anywherev1.MapRoles{
+								{
+									RoleARN:  "test-role-arn",
+									Username: "test",
+									Groups:   []string{"group1", "group2"},
+								},
+							},
+							MapUsers: []anywherev1.MapUsers{
+								{
+									UserARN:  "test-user-arn",
+									Username: "test",
+									Groups:   []string{"group1", "group2"},
+								},
+							},
+							Partition: "aws",
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			testName: "different namespace",
+			config: &cluster.Config{
+				Cluster: &anywherev1.Cluster{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       anywherev1.ClusterKind,
+						APIVersion: anywherev1.SchemeBuilder.GroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "eksa-unit-test",
+						Namespace: "default",
+					},
+					Spec: anywherev1.ClusterSpec{
+						IdentityProviderRefs: []anywherev1.Ref{
+							{
+								Name: "test1",
+								Kind: "AWSIamConfig",
+							},
+						},
+					},
+				},
+				AWSIAMConfigs: map[string]*anywherev1.AWSIamConfig{
+					"test": {
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "AWSIamConfig",
+							APIVersion: anywherev1.SchemeBuilder.GroupVersion.String(),
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "eksa-unit-test",
+							Namespace: "not-default",
+						},
+						Spec: anywherev1.AWSIamConfigSpec{
+							AWSRegion:   "test-region",
+							BackendMode: []string{"mode1", "mode2"},
+							MapRoles: []anywherev1.MapRoles{
+								{
+									RoleARN:  "test-role-arn",
+									Username: "test",
+									Groups:   []string{"group1", "group2"},
+								},
+							},
+							MapUsers: []anywherev1.MapUsers{
+								{
+									UserARN:  "test-user-arn",
+									Username: "test",
+									Groups:   []string{"group1", "group2"},
+								},
+							},
+							Partition: "aws",
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			testName: "no awsiam config",
+			config: &cluster.Config{
+				Cluster: &anywherev1.Cluster{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       anywherev1.ClusterKind,
+						APIVersion: anywherev1.SchemeBuilder.GroupVersion.String(),
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "eksa-unit-test",
+						Namespace: "default",
+					},
+					Spec: anywherev1.ClusterSpec{
+						IdentityProviderRefs: []anywherev1.Ref{
+							{
+								Name: "test1", Kind: "AWSIamConfig",
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			c := cluster.NewConfigManager()
+			c.RegisterValidations(
+				func(c *cluster.Config) error {
```

Comment:
This is not really testing the code, it's just duplicating it in the tests. In fact you can see in the codecov annotations that your new code is not being covered by tests.

Curious, what was the reason to do this?

If you want to use the config manager, I would suggest just using the `cluster.Validate(config)` method. That will use the default config manager that already has your validation registered.

The other way, is to make your new validation methods public and just call them directly from here, without the config manager.